builtin/providers/nsone/provider.go

```diff
@@ -32,9 +35,10 @@ func Provider() terraform.ResourceProvider {
 }
 
 func nsoneConfigure(d *schema.ResourceData) (interface{}, error) {
-	n := nsone.New(d.Get("apikey").(string))
-	n.Debug()
-	n.RateLimitStrategySleep()
+	httpClient := &http.Client{}
+	n := nsone.NewClient(httpClient, nsone.SetAPIKey(d.Get("apikey").(string)))
+	// FIXME: n.Debug()
+	// n.RateLimitStrategySleep()
```

Comment:
@josephholsten I made a PR for the fix: https://github.com/ns1/ns1-go/pull/24

The client was applying the rate limit func before checking the actual response code. I was able to reproduce divide by zero when i didnt set the apikey as an env var and received a 401(without rate limiting headers).